server/api.go

```diff
@@ -45,12 +46,56 @@ type PRDetails struct {
 	Reviews            []*github.PullRequestReview `json:"reviews"`
 }
 
+type HTTPHandlerFuncWithUser func(w http.ResponseWriter, r *http.Request, userID string) func()
```

Comment:
Can this be simplified even more to  
```suggestion
type HTTPHandlerFuncWithUser func(w http.ResponseWriter, r *http.Request, userID string)
```
?

Then https://github.com/bakurits/mattermost-plugin-github/blob/b4f906611637384984d941bdfcf6ca81b0537755/server/api.go#L112-L114 could also be simplified to 
```go
func (p *Plugin) connectUserToGitHub(w http.ResponseWriter, r *http.Request, userID string) {
	privateAllowed := false
```